src/lexer/scanner.go

```diff
@@ -511,8 +511,18 @@ func (s *Scanner) Scan() Token {
 				continue
 			}
 
-			errorhandling.NewLexicalError(s.currentLineFile, s.currentColumnFile, string(s.lexemBuffer)+string(currChar))
+			if len(string(s.lexemBuffer)) == 0 {
+				errorhandling.NewLexicalError(s.currentLineFile, s.currentColumnFile, string(currChar))
+			}
+
+			if len(string(s.lexemBuffer)) > 0 {
+				errorhandling.NewLexicalError(s.currentLineFile, s.currentColumnFile, string(s.lexemBuffer))
+			}
```

Comment:
The only problem with getting rid of this if statement, is that we would show 2 error log messages when the lexembuffer is empty, because the flow of the program needs to continue until it reaches the ERROR_TOKEN return.

Reply:
I see, so lets simplify with an if-else statement, what do you think?

Reply:
I agree.